### PR TITLE
Bandit plugin

### PR DESCRIPTION
**Pull request by CermakM**
Bandit is Python AST-based static analyzer from OpenStack Security Group

- find common security issues in Python code
- report generated in format compatible with csmock

- Git Hub repository: https://github.com/openstack/bandit

new file:   py/plugins/bandit.py
new file:   scripts/run-bandit.sh

Signed-off-by: Marek Cermak <macermak@redhat.com>

### CODE REVIEW

File: py/plugins/bandit.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# Copyright (C) 2014 Red Hat, Inc.
```
Comment (kdudka) [inline py/plugins/bandit.py]: s/2014/2017/

File: py/plugins/bandit.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with csmock.  If not, see <http://www.gnu.org/licenses/>.
+
+import csmock.common.util
+
+
+run_bandit_sh = "/usr/share/csmock/scripts/run-bandit.sh"
```
Comment (kdudka) [inline py/plugins/bandit.py]: The global constants need to be written in uppercase to silence pylint.  See abd9d5729c9b2c1e9da2ca9c24b8869c681fb84c.

File: scripts/run-bandit.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+SEC_LEVEL=$1
+TARGET_PATH=${@:2:$#}
+
+# Debug
+echo "[DEBUG]	SEC_LEVEL = $SEC_LEVEL" >&2
+echo "[DEBUG]	TARGET_PATH = $TARGET_PATH" >&2
+
+if [[ -z "${SEC_LEVEL// }" ]]; then
+	echo "[ERROR]	SEC_LEVEL parameter is empty"
+elif [[ -z "${SEC_LEVEL// }" ]]; then
```
Comment (kdudka) [inline scripts/run-bandit.sh]: Copy/paste error from the previous ```if``` condition?

File: scripts/run-bandit.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	echo "[ERROR]	TARGET_PATH parameter is empty"
+fi
+
+# Make sure path to bandit is present in $PATH
+export PATH="/usr/local/bin/:$PATH"
+
+# Find all python files and run bandit analysis 
+find "$TARGET_PATH" -type f -print0 \
```
Comment (kdudka) [inline scripts/run-bandit.sh]: Why does ```run-bandit.sh``` limit ```$TARGET_PATH``` to be a single directory?   ```run-pylint.sh``` takes an arbitrary count of args.

File: scripts/run-bandit.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+echo "[DEBUG]	TARGET_PATH = $TARGET_PATH" >&2
+
+if [[ -z "${SEC_LEVEL// }" ]]; then
+	echo "[ERROR]	SEC_LEVEL parameter is empty"
+elif [[ -z "${SEC_LEVEL// }" ]]; then
+	echo "[ERROR]	TARGET_PATH parameter is empty"
+fi
+
+# Make sure path to bandit is present in $PATH
+export PATH="/usr/local/bin/:$PATH"
```
Comment (kdudka) [inline scripts/run-bandit.sh]: Is this still needed?  ```py/plugins/bandit.py``` contains ```props.install_pkgs += ["bandit"]```, which should ensure that bandit is installed in default system path.

File: py/plugins/bandit.py (hunk 1 above)
Comment (CermakM) [reply py/plugins/bandit.py]: Thanks for noticing, will change

File: py/plugins/bandit.py (hunk 2 above)
Comment (CermakM) [reply py/plugins/bandit.py]: Corrected, totally forgot.

File: scripts/run-bandit.sh (hunk 5 above)
Comment (CermakM) [reply scripts/run-bandit.sh]: Oops, This is my bad here regarding run-bandit.sh .. I focused on bandit.py and forgot about this script.. It is no longer needed, will be fixed in the next commit along with errors in here.

File: scripts/run-bandit.sh (hunk 4 above)
Comment (CermakM) [reply scripts/run-bandit.sh]: Great observation! It surely is not supposed to be a single directory, the`""` are misplaced here.

File: scripts/run-bandit.sh (hunk 3 above)
Comment (CermakM) [reply scripts/run-bandit.sh]: Indeeed...

**Comment by kdudka** [comment]
Merging now.  Thanks for the contribution and sorry for the delay!

**Comment by CermakM** [comment]
No worries, Thank you!

**Pull request closed without merging**